stan-native/src/main/java/org/ballerinalang/nats/observability/NatsMetricsReporter.java

```diff
@@ -45,12 +47,23 @@ public NatsMetricsReporter(StreamingConnection connection) {
      *
      * @param url URL of the NATS server that the client is connecting to.
      */
-    public static void reportNewConnection(String url) {
+    public static void reportNewConnection(Object url) {
         if (!ObserveUtils.isMetricsEnabled()) {
             return;
         }
-        incrementGauge(new NatsObserverContext(url), NatsObservabilityConstants.METRIC_CONNECTIONS[0],
-                       NatsObservabilityConstants.METRIC_CONNECTIONS[1]);
+        if (TypeUtils.getType(url).getTag() == TypeTags.ARRAY_TAG) {
+            // if string[]
+            String[] serverUrls = ((BArray) url).getStringArray();
+            for (String serverUrl: serverUrls) {
+                incrementGauge(new NatsObserverContext(serverUrl), NatsObservabilityConstants.METRIC_CONNECTIONS[0],
```

Comment:
Shall we validate the `serverUrls` array length to avoid unexpected index outbound errors?